### DISCUSSION TITLE
"Examples" documentation ?

### DISCUSSION THREAD

**Discussion by bsimmo** [raspberrypi/picamera2#1178] [General]
Is there a list of what the examples are (in the examples folder).
What they do, and which is the important part?

I look at a big long list and really they are just random names, I click on a few and there are no comments, nor an introduction about what it is trying to show or do.

Have I missed something / some obvious place I should be looking. I checked the Wiki but nothing there, and there is no page in the examples to help out.
Thanks.

----

Something like the old and excellent _picamera_ documentation. This was really nice and easy to use on anything. https://picamera.readthedocs.io/en/release-1.13/ 
I tried the 'pdf' you link to but that's a scrollable tiny text inaccessible nightmare but doesn't really seem to help with the examples.

**Comment by davidplowman** [comment]
Hi, and thanks for the question. The examples are mostly supposed to be sufficiently straightforward that they are easy to understand, but of course maybe that's not always the case. I'd be happy to try and add more comments for any that seem difficult.

I'm not quite sure how one would turn the examples from a long list into something more searchable. Subdividing further into sub-folders sounds wrong, because many examples show more than one thing. Maybe a kind of example vs. feature matrix would be helpful? Perhaps it could live as a readme in the examples folder? There'd be the small job of deciding what meaningful user-facing features there are...!

The manual doesn't talk about the examples, it was never meant to. I'm afraid I can't really answer general questions about the format of documentation.
